azurerm/resource_arm_recovery_services_fabric.go

```diff
@@ -0,0 +1,145 @@
+package azurerm
+
+import (
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/recoveryservices/mgmt/2018-01-10/siterecovery"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmRecoveryServicesFabric() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmRecoveryServicesFabricCreate,
+		Read:   resourceArmRecoveryServicesFabricRead,
+		Update: nil,
+		Delete: resourceArmRecoveryServicesFabricDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"recovery_vault_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+			"location": azure.SchemaLocation(),
+		},
+	}
+}
+
+func resourceArmRecoveryServicesFabricCreate(d *schema.ResourceData, meta interface{}) error {
+	resGroup := d.Get("resource_group_name").(string)
+	vaultName := d.Get("recovery_vault_name").(string)
+	location := azure.NormalizeLocation(d.Get("location").(string))
+	name := d.Get("name").(string)
+
+	client := meta.(*ArmClient).recoveryServices.FabricClient(resGroup, vaultName)
+	ctx := meta.(*ArmClient).StopContext
+
+	if requireResourcesToBeImported && d.IsNewResource() {
+		existing, err := client.Get(ctx, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for presence of existing recovery services fabric: %+v", err)
```

Comment:
For this (and all other error messages) could e be a little more clear about what is erroring:
```suggestion
				return fmt.Errorf("Error checking for presence of existing recovery services fabric %s (vault %s): %+v", name, vaultName, err)
```